packages/flutter/lib/src/widgets/animated_grid.dart

```diff
@@ -0,0 +1,714 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'scroll_controller.dart';
+import 'scroll_physics.dart';
+import 'scroll_view.dart';
+import 'sliver.dart';
+import 'ticker_provider.dart';
+
+/// Signature for the builder callback used by [AnimatedGrid].
+typedef AnimatedGridItemBuilder = Widget Function(BuildContext context, int index, Animation<double> animation);
+
+/// Signature for the builder callback used by [AnimatedGridState.removeItem].
+typedef AnimatedGridRemovedItemBuilder = Widget Function(BuildContext context, Animation<double> animation);
+
+// The default insert/remove animation duration.
+const Duration _kDuration = Duration(milliseconds: 300);
+
+// Incoming and outgoing AnimatedGrid items.
+class _ActiveItem implements Comparable<_ActiveItem> {
+  _ActiveItem.incoming(this.controller, this.itemIndex) : removedItemBuilder = null;
+  _ActiveItem.outgoing(this.controller, this.itemIndex, this.removedItemBuilder);
+  _ActiveItem.index(this.itemIndex)
+      : controller = null,
+        removedItemBuilder = null;
+
+  final AnimationController? controller;
+  final AnimatedGridRemovedItemBuilder? removedItemBuilder;
+  int itemIndex;
+
+  @override
+  int compareTo(_ActiveItem other) => itemIndex - other.itemIndex;
+}
+
+/// A scrolling container that animates items when they are inserted or removed
+/// in a grid.
+///
+/// This widget's [AnimatedGridState] can be used to dynamically insert or
+/// remove items. To refer to the [AnimatedGridState] either provide a
+/// [GlobalKey] or use the static [of] method from an item's input callback.
+///
+/// This widget is similar to one created by [GridView.builder].
+///
+/// {@tool dartpad}
+/// This sample application uses an [AnimatedGrid] to create an effect when
+/// items are removed or added to the grid.
+///
+/// ** See code in examples/api/lib/widgets/animated_grid/animated_grid.0.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+/// * [SliverAnimatedGrid], a sliver which animates items when they are inserted
+///   or removed from a grid.
+/// * [SliverAnimatedList], a sliver which animates items added and removed from
+///   a list instead of a grid.
+/// * [AnimatedList], which animates items added and removed from a list instead
```

Comment:
Can you add `see also` to the AnimatedList/AnimatedSliverLists that points to these grid versions?